common/tools/eslint-plugin-azure-sdk/src/rules/ts-package-json-name.ts

```diff
@@ -63,14 +65,32 @@ export = {
                     "service name matches directory name, but the directory is not kebab case (lowercase and separated by hyphens)"
                 });
               }
-            } else if (name !== `@azure/${packageDirectory}`) {
+            } else if (!isValidFolder(name, packageDirectory)) {
+              const subScopeSuffix = subScope ?? "";
               context.report({
                 node: nodeValue,
-                message: `service should be named '@azure/${packageDirectory}' or should be moved to a directory called '${packageBaseName}'`
+                message: `service should be named '@azure${subScopeSuffix}/${packageDirectory}' or should be moved to a directory called '${packageBaseName}${subScopeSuffix}'`
               });
             }
           }
         } as Rule.RuleListener)
       : {};
   }
 };
+
+function isValidFolder(packageName: string, folderName: string): boolean {
+  const subScope = getSubScope(packageName);
+  if (!subScope) {
+    return RegExp(`^@azure(-[a-z]+)?\/${folderName}`).test(packageName);
+  }
+
+  // If there is a subScope, allow the  folder name to have it appended at the end i.e folder-name-subscope
+  return RegExp(`^@azure(-[a-z]+)?\/${folderName}`).test(`${packageName}${subScope}`);
+}
+
+function getSubScope(packageName: string): string | undefined {
+  // Check if there is a sub scope i.e @azure-rest
+  const [_, subScope] = packageName.match(/^@azure(-[a-z]+)?\//) ?? [];
+
+  return subScope;
```

Comment:
It might be better to repurpose this function to return both the subscope and the package name in one pass. Perhaps, it will need to be renamed to `getPackageMetadata` or something like that.